foundation/foundation-mda/src/test/java/com/boozallen/aiops/mda/generator/MlPipelineStep.java

```diff
@@ -0,0 +1,185 @@
+    public void a_machine_learning_pipeline_with_missing_step_module() throws IOException {
+        createProject("ml-pipeline-test", "step-1");
+        createMlPipelineWithMultipleSteps();
+    }
+
+    @When("the aissemble-maven-modules profile is generated")
+    public void the_aissemble_maven_modules_profile_is_generated() throws Exception {
+        generateProfile("aissemble-maven-modules", this::createPipelinesRootPomContext);
+    }
+
+    @When("the machine-learning-pipeline profile is generated")
+    public void the_machine_learning_pipeline_profile_is_generated() throws Exception {
+        generateProfile("machine-learning-pipeline", this::createGenerationContext);
+    }
+
+    @Then("the pipeline POM is generated under the pipeline module")
+    public void the_pipeline_pom_is_generated_under_the_pipeline_module() throws IOException {
+        assertPomExists("the pipeline POM is generated under the pipeline module", PIPELINE_NAME);
+    }
+
+    @Then("a POM is generated for each step under the pipeline module")
+    public void a_pom_is_generated_for_each_step_under_the_pipeline_module() throws IOException {
+        for (String step: STEPS) {
+            assertPomExists(String.format("%s module is generated under the pipeline module", step), PIPELINE_NAME, step);
+        }
+    }
+
+    @Then("the step modules are listed in the pipeline POM")
+    public void the_step_modules_are_listed_in_the_pipeline_pom() throws IOException {
+        Path pom = projectDir.resolve(PIPELINE_NAME).resolve("pom.xml");
+        assertModulesExistInPom(pom, STEPS, "The step modules are listed in the pipeline POM");
+    }
+
+    @Then("a POM is generated for new step under the pipeline module")
+    public void a_pom_is_generated_for_new_step_under_the_pipeline_module() throws Exception {
+        assertPomExists(String.format("%s module is generated under the pipeline module", NEW_STEP), PIPELINE_NAME, NEW_STEP);
+    }
+
+    @Then("a Manual Action is prompted for adding the step module to the pipeline")
+    public void a_manual_action_is_prompted_for_adding_the_step_module_to_the_pipeline() throws IOException {
+        Map<String, Map<String, Notification>> notifications = NotificationCollector.getNotifications();
+        List<String> mlStepsInManualAction = new ArrayList<>();
+
+        notifications.values().forEach(item -> {
+            item.values().forEach(notification -> {
+                for (String step: STEPS) {
+                    if (notification.getItems().contains(String.format("<module>%s</module>", step))) {
+                        mlStepsInManualAction.add(step);
+
+                        if (mlStepsInManualAction.size() == STEPS.size()) {
+                            break;
+                        }
+                    }
+                }
+            });
+        });
+        assertTrue("Missing steps are included in the Manual Action Prompt", mlStepsInManualAction.size() == STEPS.size());
```

Comment:
A: I'd include expected & found steps in the failure statement to help with debugging.